src/convertor_tests.rs

```diff
@@ -255,55 +254,60 @@ fn extract_simulator_config_from_revlog() {
         .iter()
         .filter(|r| r.review_kind == RevlogReviewKind::Learning && r.ease_factor == 0)
         .counts_by(|r| r.button_chosen);
-    let first_rating_prob = first_rating_count
-        .iter()
-        .map(|(button_chosen, count)| {
-            (
-                *button_chosen,
-                *count as f32 / first_rating_count.values().sum::<usize>() as f32,
-            )
-        })
-        .collect::<HashMap<_, _>>();
-    let first_rating_prob = [
-        first_rating_prob.get(&1).copied().unwrap_or_default(),
-        first_rating_prob.get(&2).copied().unwrap_or_default(),
-        first_rating_prob.get(&3).copied().unwrap_or_default(),
-        first_rating_prob.get(&4).copied().unwrap_or_default(),
-    ];
-    assert_eq!(first_rating_prob, [0.15339181, 0.0, 0.15339181, 0.6932164,]);
+    let first_rating_prob = {
+        let mut arr = [Default::default(); 5];
+        first_rating_count
+            .iter()
+            .map(|(button_chosen, count)| {
+                (
+                    *button_chosen,
+                    *count as f32 / first_rating_count.values().sum::<usize>() as f32,
+                )
+            })
+            .for_each(|(i, v)| {
+                arr[i as usize] = v;
+            });
+        arr
+    };
+    assert_eq!(
+        first_rating_prob[1..=4],
```

Comment:
Might be simpler to .filter() out the button_chosen=0 case

Reply:
How about:

            .for_each(|(i, v)| {
                if i > 0 { arr[i-1 as usize] = v; }
            });